waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -487,7 +491,7 @@ proc handleMessage*(w: WakuStore, topic: string, msg: WakuMessage) {.async.} =
 
   let res = w.store.put(index, msg, topic)
   if res.isErr:
-    warn "failed to store messages", err = res.error
+    trace "failed to store messages", err = res.error
```

Comment:
> I see the implementation is read through only, this might lead to data inconsistency and i don't see any timestamp or write through strategies, is our assumption messages are not to be modified once in store?

I may not get your exact meaning, but when we query/retrieve data we will only ever consult our in-memory store/cache. This will never in turn query the persistent data storage (we _only_ read from this during startup), which to me is not quite a "read through" strategy? For each new message we receive, we'll store a copy in-memory and on disk (somewhat of a "write through" strategy). The disk/persistent copy is only meant to serve as backup to prime the store after restarts. Every query (read/write) will primarily interface with the in-memory store. Not sure if that answers your question? Also, very happy to receive any suggestions here if there are ways we can do it better - I already think we may have to do batched reads/writes to persistent storage, but that's for a follow-up PR. LMK what you think. :)

Reply:
> is there a reason store failure message is trace?

Whenever we enter a failure condition, this will be logged _per message_. At high message rates the act of logging itself can cause further processing bottle necks and make the logs unusable, as we have indeed seen happening. Even `debug` level has caused such log overflows before. One way of addressing this is to introduce some log filter that will log this at `warn` level once and at `trace`  for every subsequent, say, 10000 messages. Since we do have visibility of these errors in a metric, a first step is to remove per-message `warn` logs that are currently clogging up `store` fleet nodes.